routing/bandwidth.go

```diff
@@ -13,11 +13,20 @@ type bandwidthHints interface {
 	// channel and a bool indicating whether the channel hint was found.
 	// If the channel is unavailable, a zero amount is returned.
 	availableChanBandwidth(channelID uint64) (lnwire.MilliSatoshi, bool)
+
+	// availableHtlcBandwidth returns the total available bandwidth that
```

Comment:
Do we actually need _both_ of these? (still getting through the diff)

Reply:
Could def cut down and just have one method that takes a pointer to amount (if present), I thought it was nicer from the caller's perspective to not need to care about that detail.

Reply:
Seems duplicate here. I think what we want here,
1. know the current bandwidth
2. check if our bandwidth can carry the amount

Step 2 is implemented in both `getPolicyLocal` and `MayAddOutgoingHtlc`, sort redundant?